File: send-data-from-one-fragment-to-another-java-version/app/src/main/java/com/example/senddatafromonefragmmenttoanotherjavaversion/MainActivity.java
package com.example.senddatafromonefragmmenttoanotherjavaversion;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity implements Communicator{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FirstFragment firstFragment = new FirstFragment();
        getSupportFragmentManager().beginTransaction().replace(R.id.relativeLayout, firstFragment).commit();

    }

    @Override
    public void passData(String editTextInput) {
        Bundle bundle = new Bundle();
        bundle.putString("inputText", editTextInput);
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        FragmentTwo fragmentTwo = new FragmentTwo();

        fragmentTwo.setArguments(bundle);
        fragmentTransaction.replace(R.id.relativeLayout, fragmentTwo);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
        fragmentTransaction.commit();
    }
}

File: send-data-from-one-fragment-to-another-fragment-using-kotlin/app/src/main/java/com/example/senddatafromonefragmenttoanotherusingkotlin/MainActivity.kt
package com.example.senddatafromonefragmenttoanotherusingkotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentTransaction

class MainActivity : AppCompatActivity(), Communicator {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val firstFragment = FirstFragment()
        supportFragmentManager.beginTransaction().replace(R.id.relativeLayout, firstFragment)
            .commit()
    }

    override fun passData(editTextInput: String) {
        val bundle = Bundle()
        bundle.putString("inputText", editTextInput)
        val transaction = this.supportFragmentManager.beginTransaction()
        val fragmentTwo = FragmentTwo()

        fragmentTwo.arguments = bundle
        transaction.replace(R.id.relativeLayout, fragmentTwo)
        transaction.addToBackStack(null)
        transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
        transaction.commit()
    }
}